app/controllers/application_controller.rb

```diff
@@ -22,4 +22,15 @@ def current_admin?
   def current_merchant?
     current_user && current_user.merchant?
   end
+
+  def redirect_based_on_role(role)
```

Comment:
Nice use of a helper method here since we will be using the same logic across two controllers